src/vscode-dts/vscode.d.ts

```diff
@@ -19631,6 +19680,40 @@ declare module 'vscode' {
 		 * @returns An array of chat models, can be empty!
 		 */
 		export function selectChatModels(selector?: LanguageModelChatSelector): Thenable<LanguageModelChat[]>;
+
+		/**
+		 * Register a LanguageModelTool. The tool must also be registered in the package.json `languageModelTools` contribution
+		 * point. A registered tool is available in the {@link lm.tools} list for any extension to see. But in order for it to
+		 * be seen by a language model, it must be passed in the list of available tools in {@link LanguageModelChatRequestOptions.tools}.
+		 * @returns A {@link Disposable} that unregisters the tool when disposed.
+		 */
+		export function registerTool<T>(name: string, tool: LanguageModelTool<T>): Disposable;
+
+		/**
+		 * A list of all available tools that were registered by all extensions using {@link lm.registerTool}. They can be called
+		 * with {@link lm.invokeTool} with a set of parameters that match their declared `parametersSchema`.
+		 */
+		export const tools: readonly LanguageModelToolInformation[];
+
+		/**
+		 * Invoke a tool listed in {@link lm.tools} by name with the given parameters.
+		 *
+		 * The caller must pass a {@link LanguageModelToolInvocationOptions.toolInvocationToken}, which comes from
+		 * {@link ChatRequest.toolInvocationToken} when the tool is being invoked by a by a {@link ChatParticipant}, and
+		 * associates the invocation to a chat session.
+		 *
+		 * The tool will return a {@link LanguageModelToolResult} which contains an array of {@link LanguageModelTextPart} and
+		 * {@link LanguageModelPromptTsxPart}. If the tool caller is using `@vscode/prompt-tsx`, it can incorporate the response
+		 *  parts into its prompt using a `ToolResult`. If not, the parts can be passed along to the {@link LanguageModelChat} via
+		 *  a User message with a {@link LanguageModelToolResultPart}.
+		 *
+		 * If a chat participant wants to preserve tool results for requests across multiple turns, it can store tool results in
+		 * the {@link ChatResult.metadata} returned from the handler and retrieve them on the next turn from
+		 * {@link ChatResponseTurn.result}.
+		 *
+		 * @returns The result of the tool invocation.
```

Comment:
mention how/if parameter are validated, like "this call will fail if the the passed options don't match the corresponding schema"

Reply:
I couldn't decide how to phrase it since I don't do that yet, but I'll mention something